llvm/lib/Transforms/Instrumentation/RealtimeSanitizer.cpp

```diff
@@ -0,0 +1,50 @@
+#include "llvm/IR/Analysis.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Module.h"
+
+#include "llvm/Transforms/Instrumentation/RealtimeSanitizer.h"
+
+using namespace llvm;
+
+namespace {
+
+void insertCallBeforeInstruction(Function &Fn, Instruction &Instruction,
+                                 const char *FunctionName) {
```

Comment:
Do you expect Module specific features? Other sanitizers are ModulePasses already.

Reply:
At this time, no we do not expect to have Module specific features. A realtime or "nonblocking" context only deals with functions. We will insert appropriate calls at points within each function.

 As far as I can tell, the FunctionPass will be all that we need.

Reply:
Prefer static for internal linkage functions
https://llvm.org/docs/CodingStandards.html#anonymous-namespaces